src/Compilers/CSharp/Portable/CodeGen/EmitStackAllocInitializer.cs

```diff
@@ -56,22 +78,74 @@ private void EmitStackAllocInitializers(TypeSymbol type, BoundArrayInitializatio
                     _builder.EmitToken(field, inits.Syntax, _diagnostics.DiagnosticBag);
                     _builder.EmitIntConstant(data.Length);
                     _builder.EmitOpCode(ILOpCode.Cpblk, -3);
-
-                    if (initializationStyle == ArrayInitializerStyle.Mixed)
+                }
+                else
+                {
+                    if (getCreateSpanHelper(_module, elementType) is { } createSpanHelper &&
+                        _module.Compilation.GetWellKnownTypeMember(WellKnownMember.System_ReadOnlySpan_T__GetPinnableReference) is MethodSymbol getPinnableReference)
                     {
-                        EmitElementStackAllocInitializers(elementType, initExprs, includeConstants: false);
+                        // Use RuntimeHelpers.CreateSpan and cpblk.
+                        EmitStackAllocBlockMultiByteInitializer(data, createSpanHelper, getPinnableReference, elementType, inits.Syntax, _diagnostics.DiagnosticBag);
```

Comment:
>EmitStackAllocBlockMultiByteInitializer(data, createSpanHelper, getPinnableReference, elementType, inits.Syntax, _diagnostics.DiagnosticBag) [](http://example.com/codeflow?start=24&length=140)

How do we know that the element type is actually suitable for that?

Reply:
Are we testing the fact?

Reply:
It looks like ```ShouldEmitBlockInitializerForStackAlloc``` relies on ```IsBlittable``` , but ```TryEmitReadonlySpanAsBlobWrapper``` and ```ShouldUseRuntimeHelpersCreateSpan``` are relying on ```IsTypeAllowedInBlobWrapper``` helper to decide if `CreateSpan` can be used for the type. I think we should either align all these places to use the same way of checking for acceptable types (assuming the checks are equivalent), or, perhaps, call ```IsTypeAllowedInBlobWrapper``` here explicitly as an extra condition.